Add help/get-started pages

clhenrick opened the pull request:
This PR supersedes #4 as it does a better job of utilizing the existing patterns in `src/routes/help`.

- [x] Getting Started landing page
- [x] Getting Started topic pages
- [x] Adds topic pages to left sidebar navigation
- [ ] ~~Adds topic page anchors to right sidebar navigation~~
- [x] Adds previous and next topic links to bottom of topic pages

clhenrick committed:
updated content for get-started/1

clhenrick committed:
start stubbing out get-started route

clhenrick committed:
added support footer component

clhenrick committed:
added module for help category icons

clhenrick committed:
get started uses category icons

clhenrick committed:
added css custom properties for all sass variables

clhenrick committed:
renamed component file name for SidebarLeft to work on linux paths

clhenrick committed:
render cards for each getting-started topic

clhenrick committed:
abstracted TopicTile component

clhenrick committed:
fixed typo of _SidebarLeft

clhenrick committed:
updated styling & markup for TopicCard component

clhenrick committed:
added intro paragraph

clhenrick committed:
added dev dep: html-loader

clhenrick committed:
updated webpack config to handle loading html files

clhenrick committed:
added html file to test

clhenrick committed:
render test html file in get-started/:slug

clhenrick committed:
fix alias in webpack config

clhenrick committed:
added descriptions for get-started topics

clhenrick committed:
more layout for topic pages

clhenrick committed:
updated SidebarRight component to use page anchor links

clhenrick committed:
topic pages use SidebarRight with page anchor link

clhenrick committed:
updated SupportFooter: add optional top border

clhenrick committed:
minor cleanup

clhenrick committed:
tmp hide get-started dir from sapper

clhenrick committed:
updated grids sass partial: add .content-extended class

clhenrick committed:
render get-started landing page in [category].svelte

clhenrick committed:
updated markdown files for get-started pages

clhenrick committed:
added default table styling in _global.scss

clhenrick committed:
use get-started topics data in help/categories/slug.svelte

clhenrick commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Help: Get Started](https://trello.com/c/2aKsnNzX/15-help-get-started)

clhenrick committed:
clean up old files in help/_get-started dir

clhenrick committed:
use SupportFooter component in [category]/[slug].svelte

clhenrick commented:
@mukhtyar folliwing up on our discussion today, these were the following things I'm curious about doing with the Getting Started topic pages and Sapper:

- indented links in the left side nav for the getting started topic pages
- page anchor links for headings in each getting started topic page
- previous and next links at the bottom of an article for moving between topic pages

But if we are planning on migrating to Svelte Kit soon then maybe it's not worth looking into too much right now? So please don't feel like you need to spend too much time on it or feel free to put it off till after we migrate. Thank you!

mukhtyar commented:
When you land on `/help/get-started/` you have access to 2 variables `toc` and `data`. Currently only the `toc` variable is passes in as prop to `_SidebarLeft.svelte` component. I think if you pass in the `data` as prop, you should have access to the list of items under `get-started`.

In `_SidebarLeft.svelte` if you uncomment the lines 4-5 and 22-34 and update the `if` condition you will see indented links under Get Started.

mukhtyar commented:
To add Previous and Next links at bottom of each page, update the `_ItemDetail.svelte` with new props and pass in the previous and next urls. This will add it to any content that uses `_ItemDetail.svelte`, at this point I think it's items under FAQ and Get Started.

mukhtyar commented:
We can skip page anchor links for headings in each getting started topic page for now. [mdsvex](https://mdsvex.com/docs) seems like a more powerful markdown parser, we can switch to that when we migrate to svelte-kit.

clhenrick commented:
thanks for the info @mukhtyar, that sounds good to me.

clhenrick committed:
rm topic page heading anchors

clhenrick committed:
implemented nested nav items in SidebarLeft

clhenrick committed:
minor adjustment to SidebarLeft sub nav styling

clhenrick committed:
a11y fix for SidebarLeft nav: add aria-label

clhenrick committed:
update markdown files: rm page anchors &  "questions"

clhenrick committed:
(wip: add prev & next article links

clhenrick committed:
fixed bug with subToc index

clhenrick committed:
improve a11y & cleaned up css in ItemsPrevNextNav

clhenrick committed:
renamed component ItemsNavPrevNext

clhenrick commented:
@mukhtyar this one is ready for your review, no rush! 

Some brief notes:

- I added an alias in the webpack config for the `src/` directory, so instead of navigating up a path like `../../../dir` you can just do `~/dir`
- the `_getting-started-topics.js` contains data similar to `_categories.js` but I omitted the `dir` prop since these files will always come from `content/get-started/`
- I made the footer that has the support email contact its own component
- I added a few other components: Get Started, TopicTile, ItemNavPrevNext
- The `ItemNavPrevNext` component could also be used for the FAQs page if we want
- I tried to update the `SidebarLeft` so that it isn't hardcoded to `get-started` in case we want to use the nested TOC pattern for other sections in Help.

Let me know if you have any questions or suggestions, happy to jump on a quick Zoom call or talk over chat. Thanks!

clhenrick committed:
Merge branch 'main' into clhenrick--get-started2

clhenrick committed:
rm unused file in content/get-started dir

mukhtyar commented on src/partials/SidebarRight.svelte:
```diff
@@ -1,11 +1,15 @@
   export let events = [];
   export let posts = [];
   export let filters = [];
+  export let anchors = [];
```
So any sub-headings will show up in the right sidebar? I like that idea.

clhenrick replied:
Thanks @mukhtyar and yes, that was my thinking. It could be helpful for longer articles with many headings > h1 but we might want to disable it for articles that have only a single h2 heading.

mukhtyar commented on src/routes/help/[category]/[slug].svelte:
```diff
@@ -97,13 +122,15 @@
       <ItemDetail {item} />
+      <Row>
+        <Column>
+          <ItemsPrevNextNav prev={prevArticle} next={nextArticle} />
```
Adding it to FAQs might be useful too.

mukhtyar commented on src/routes/help/_TopicTile.svelte:
```diff
@@ -0,0 +1,87 @@
+<style lang="scss">
```
Just realized the prettier config has `svelteSortOrder: "styles-markup-scripts"` and I've been doing `"scripts-styles-markup"` order. We should probably decide on one.

clhenrick replied:
Ah that's why that's happening! I thought it was just my VS Code settings so was trying not re-format any existing files. I'm not partial to either, perhaps it makes sense to use `"scripts-styles-markup"` as that's the pattern used through the majority of the Svelte files.

mukhtyar reviewed (approved):
Looks great! We should talk about some common eslint rules :-)

clhenrick commented:
Thanks @mukhtyar for the review! I'll add a Trello card for us to follow up on the ES Lint & Prettier configurations and one for adding the previous & next article links to the FAQs page.
